### ISSUE TITLE
getting "attempt to compare number with userdata" error

### ISSUE THREAD

**Issue by ghost** [Kong/kong#4412]
### Summary

Suddenly all my 'routes' disappeared this morning after Kong had been running normally for four days. I was able to add 'service' but couldn't add 'route', kong returned 'An unexpected error occurred' when I added 'route' to it.  Then Kong returns 'attempt to compare number with userdata' error as long as I start it since I stopped it.

### Steps To Reproduce

1.kong start

### Additional Details & Logs

- Kong version (`$ kong version`)
root@ubuntu:~# kong version
Kong Enterprise 0.34-1

- Kong debug-level startup logs (`$ kong start --vv`)
2019/03/20 03:35:16 [verbose] Kong: 0.34-1-enterprise-edition
2019/03/20 03:35:16 [debug] ngx_lua: 10013
2019/03/20 03:35:16 [debug] nginx: 1013006
2019/03/20 03:35:16 [debug] Lua: LuaJIT 2.1.0-beta3
2019/03/20 03:35:16 [verbose] reading config file at /etc/kong/kong.conf
2019/03/20 03:35:16 [debug] admin_access_log = "logs/admin_access.log"
2019/03/20 03:35:16 [debug] admin_approved_email = "true"
2019/03/20 03:35:16 [debug] admin_docs_url = "https://docs.konghq.com/enterprise/0.34/admin-gui/overview/"
2019/03/20 03:35:16 [debug] admin_emails_from = "\"\""
2019/03/20 03:35:16 [debug] admin_error_log = "logs/error.log"
2019/03/20 03:35:16 [debug] admin_gui_access_log = "logs/admin_gui_access.log"
2019/03/20 03:35:16 [debug] admin_gui_error_log = "logs/admin_gui_error.log"
2019/03/20 03:35:16 [debug] admin_gui_flags = "{}"
2019/03/20 03:35:16 [debug] admin_gui_listen = {"0.0.0.0:8002","0.0.0.0:8445 ssl"}
2019/03/20 03:35:16 [debug] admin_invitation_expiry = 259200
2019/03/20 03:35:16 [debug] admin_listen = {"192.168.129.132:8001","192.168.129.132:8444 ssl"}
2019/03/20 03:35:16 [debug] anonymous_reports = true
2019/03/20 03:35:16 [debug] audit_log = false
2019/03/20 03:35:16 [debug] audit_log_ignore_methods = {}
2019/03/20 03:35:16 [debug] audit_log_ignore_paths = {}
2019/03/20 03:35:16 [debug] audit_log_ignore_tables = {}
2019/03/20 03:35:16 [debug] audit_log_record_ttl = 2592000
2019/03/20 03:35:16 [debug] cassandra_consistency = "ONE"
2019/03/20 03:35:16 [debug] cassandra_contact_points = {"127.0.0.1"}
2019/03/20 03:35:16 [debug] cassandra_data_centers = {"dc1:2","dc2:3"}
2019/03/20 03:35:16 [debug] cassandra_keyspace = "kong"
2019/03/20 03:35:16 [debug] cassandra_lb_policy = "RoundRobin"
2019/03/20 03:35:16 [debug] cassandra_port = 9042
2019/03/20 03:35:16 [debug] cassandra_repl_factor = 1
2019/03/20 03:35:16 [debug] cassandra_repl_strategy = "SimpleStrategy"
2019/03/20 03:35:16 [debug] cassandra_schema_consensus_timeout = 10000
2019/03/20 03:35:16 [debug] cassandra_ssl = false
2019/03/20 03:35:16 [debug] cassandra_ssl_verify = false
2019/03/20 03:35:16 [debug] cassandra_timeout = 5000
2019/03/20 03:35:16 [debug] cassandra_username = "kong"
2019/03/20 03:35:16 [debug] client_body_buffer_size = "8k"
2019/03/20 03:35:16 [debug] client_max_body_size = "0"
2019/03/20 03:35:16 [debug] client_ssl = false
2019/03/20 03:35:16 [debug] custom_plugins = {}
2019/03/20 03:35:16 [debug] database = "postgres"
2019/03/20 03:35:16 [debug] db_cache_ttl = 3600
2019/03/20 03:35:16 [debug] db_update_frequency = 5
2019/03/20 03:35:16 [debug] db_update_propagation = 0
2019/03/20 03:35:16 [debug] dns_error_ttl = 1
2019/03/20 03:35:16 [debug] dns_hostsfile = "/etc/hosts"
2019/03/20 03:35:16 [debug] dns_no_sync = false
2019/03/20 03:35:16 [debug] dns_not_found_ttl = 30
2019/03/20 03:35:16 [debug] dns_order = {"LAST","SRV","A","CNAME"}
2019/03/20 03:35:16 [debug] dns_resolver = {}
2019/03/20 03:35:16 [debug] dns_stale_ttl = 4
2019/03/20 03:35:16 [debug] enforce_rbac = "off"
2019/03/20 03:35:16 [debug] error_default_type = "text/plain"
2019/03/20 03:35:16 [debug] latency_tokens = true
2019/03/20 03:35:16 [debug] log_level = "notice"
2019/03/20 03:35:16 [debug] lua_package_cpath = ""
2019/03/20 03:35:16 [debug] lua_package_path = "./?.lua;./?/init.lua;"
2019/03/20 03:35:16 [debug] lua_socket_pool_size = 30
2019/03/20 03:35:16 [debug] lua_ssl_verify_depth = 1
2019/03/20 03:35:16 [debug] mem_cache_size = "128m"
2019/03/20 03:35:16 [debug] nginx_daemon = "on"
2019/03/20 03:35:16 [debug] nginx_optimizations = true
2019/03/20 03:35:16 [debug] nginx_user = "nobody nobody"
2019/03/20 03:35:16 [debug] nginx_worker_processes = "auto"
2019/03/20 03:35:16 [debug] pg_database = "kong"
2019/03/20 03:35:16 [debug] pg_host = "192.168.129.132"
2019/03/20 03:35:16 [debug] pg_password = "******"
2019/03/20 03:35:16 [debug] pg_port = 5432
2019/03/20 03:35:16 [debug] pg_ssl = false
2019/03/20 03:35:16 [debug] pg_ssl_verify = false
2019/03/20 03:35:16 [debug] pg_user = "kong"
2019/03/20 03:35:16 [debug] portal = true
2019/03/20 03:35:16 [debug] portal_access_request_email = true
2019/03/20 03:35:16 [debug] portal_api_access_log = "logs/portal_api_access.log"
2019/03/20 03:35:16 [debug] portal_api_error_log = "logs/error.log"
2019/03/20 03:35:16 [debug] portal_api_listen = {"0.0.0.0:8004","0.0.0.0:8447 ssl"}
2019/03/20 03:35:16 [debug] portal_approved_email = true
2019/03/20 03:35:16 [debug] portal_auto_approve = false
2019/03/20 03:35:16 [debug] portal_gui_host = "127.0.0.1:8003"
2019/03/20 03:35:16 [debug] portal_gui_listen = {"0.0.0.0:8003","0.0.0.0:8446 ssl"}
2019/03/20 03:35:16 [debug] portal_gui_protocol = "http"
2019/03/20 03:35:16 [debug] portal_gui_use_subdomains = false
2019/03/20 03:35:16 [debug] portal_invite_email = true
2019/03/20 03:35:16 [debug] portal_reset_email = true
2019/03/20 03:35:16 [debug] portal_reset_success_email = true
2019/03/20 03:35:16 [debug] portal_token_exp = 21600
2019/03/20 03:35:16 [debug] prefix = "/usr/local/kong/"
2019/03/20 03:35:16 [debug] proxy_access_log = "logs/access.log"
2019/03/20 03:35:16 [debug] proxy_error_log = "logs/error.log"
2019/03/20 03:35:16 [debug] proxy_listen = {"0.0.0.0:8000","0.0.0.0:8443 ssl"}
2019/03/20 03:35:16 [debug] rbac_auth_header = "Kong-Admin-Token"
2019/03/20 03:35:16 [debug] real_ip_header = "X-Real-IP"
2019/03/20 03:35:16 [debug] real_ip_recursive = "off"
2019/03/20 03:35:16 [debug] server_tokens = true
2019/03/20 03:35:16 [debug] smtp_admin_emails = {"admin@example.com"}
2019/03/20 03:35:16 [debug] smtp_domain = "localhost.localdomain"
2019/03/20 03:35:16 [debug] smtp_host = "localhost"
2019/03/20 03:35:16 [debug] smtp_mock = true
2019/03/20 03:35:16 [debug] proxy_error_log = "logs/error.log"
2019/03/20 03:35:16 [debug] proxy_listen = {"0.0.0.0:8000","0.0.0.0:8443 ssl"}
2019/03/20 03:35:16 [debug] rbac_auth_header = "Kong-Admin-Token"
2019/03/20 03:35:16 [debug] real_ip_header = "X-Real-IP"
2019/03/20 03:35:16 [debug] real_ip_recursive = "off"
2019/03/20 03:35:16 [debug] server_tokens = true
2019/03/20 03:35:16 [debug] smtp_admin_emails = {"admin@example.com"}
2019/03/20 03:35:16 [debug] smtp_domain = "localhost.localdomain"
2019/03/20 03:35:16 [debug] smtp_host = "localhost"
2019/03/20 03:35:16 [debug] smtp_mock = true
2019/03/20 03:35:16 [debug] smtp_port = 25
2019/03/20 03:35:16 [debug] smtp_ssl = false
2019/03/20 03:35:16 [debug] smtp_starttls = false
2019/03/20 03:35:16 [debug] smtp_timeout_connect = 60000
2019/03/20 03:35:16 [debug] smtp_timeout_read = 60000
2019/03/20 03:35:16 [debug] smtp_timeout_send = 60000
2019/03/20 03:35:16 [debug] ssl_cipher_suite = "modern"
2019/03/20 03:35:16 [debug] ssl_ciphers = "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256"
2019/03/20 03:35:16 [debug] trusted_ips = {}
2019/03/20 03:35:16 [debug] upstream_keepalive = 60
2019/03/20 03:35:16 [debug] vitals = true
2019/03/20 03:35:16 [debug] vitals_delete_interval_pg = 30
2019/03/20 03:35:16 [debug] vitals_flush_interval = 10
2019/03/20 03:35:16 [debug] vitals_prometheus_scrape_interval = 5
2019/03/20 03:35:16 [debug] vitals_statsd_prefix = "kong"
2019/03/20 03:35:16 [debug] vitals_statsd_udp_packet_size = 1024
2019/03/20 03:35:16 [debug] vitals_strategy = "database"
2019/03/20 03:35:16 [debug] vitals_ttl_minutes = 90000
2019/03/20 03:35:16 [debug] vitals_ttl_seconds = 3600
2019/03/20 03:35:16 [verbose] prefix in use: /usr/local/kong
2019/03/20 03:35:16 [verbose] preparing nginx prefix directory at /usr/local/kong
2019/03/20 03:35:16 [verbose] SSL enabled, no custom certificate set: using default certificate
2019/03/20 03:35:16 [verbose] default SSL certificate found at /usr/local/kong/ssl/kong-default.crt
2019/03/20 03:35:16 [verbose] Admin SSL enabled, no custom certificate set: using default certificate
2019/03/20 03:35:16 [verbose] admin SSL certificate found at /usr/local/kong/ssl/admin-kong-default.crt
2019/03/20 03:35:16 [verbose] Admin GUI SSL enabled, no custom certificate set: using default certificate
2019/03/20 03:35:16 [verbose] admin_gui SSL certificate found at /usr/local/kong/ssl/admin-gui-kong-default.crt
2019/03/20 03:35:16 [verbose] Developer Portal GUI SSL enabled, no custom certificate set: using default certificate
2019/03/20 03:35:16 [verbose] portal_gui SSL certificate found at /usr/local/kong/ssl/portal-gui-kong-default.crt
2019/03/20 03:35:16 [verbose] Developer Portal API SSL enabled, no custom certificate set: using default certificate
2019/03/20 03:35:16 [verbose] portal_api SSL certificate found at /usr/local/kong/ssl/portal-api-kong-default.crt
2019/03/20 03:35:16 [debug] searching for OpenResty 'nginx' executable
2019/03/20 03:35:16 [debug] /usr/local/openresty/nginx/sbin/nginx -v: 'nginx version: openresty/1.13.6.2'
2019/03/20 03:35:16 [debug] found OpenResty 'nginx' executable at /usr/local/openresty/nginx/sbin/nginx
2019/03/20 03:35:16 [debug] starting nginx: /usr/local/openresty/nginx/sbin/nginx -p /usr/local/kong -c nginx.conf
2019/03/20 03:35:16 [verbose] could not start Kong, stopping services
2019/03/20 03:35:16 [verbose] stopped services

- Kong error logs (`<KONG_PREFIX>/logs/error.log`)
root@ubuntu:~# vim /usr/local/kong/logs/error.log
2019/03/14 08:34:53 [notice] 5542#0: using the "epoll" event method
2019/03/14 08:34:53 [notice] 5542#0: openresty/1.13.6.2
2019/03/14 08:34:53 [notice] 5542#0: built by gcc 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.4)
2019/03/14 08:34:53 [notice] 5542#0: OS: Linux 4.15.0-46-generic
2019/03/14 08:34:53 [notice] 5542#0: getrlimit(RLIMIT_NOFILE): 1024:1048576
2019/03/14 08:34:53 [notice] 5560#0: start worker processes
2019/03/14 08:34:53 [notice] 5560#0: start worker process 5561
2019/03/14 08:34:53 [notice] 5560#0: start worker process 5562
2019/03/14 08:34:53 [notice] 5560#0: start worker process 5563
2019/03/14 08:34:53 [notice] 5560#0: start worker process 5564
2019/03/14 08:34:53 [error] 5561#0: *7 [lua] license_helpers.ljbc:0: (): The Kong Enterprise license will expire on 2019-04-11. Please contact <support@konghq.com> to renew your license., context: ngx.timer
2019/03/14 08:34:53 [notice] 5542#0: using the "epoll" event method
2019/03/14 08:34:53 [notice] 5542#0: openresty/1.13.6.2
2019/03/14 08:34:53 [notice] 5542#0: built by gcc 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.4)
2019/03/14 08:34:53 [notice] 5542#0: OS: Linux 4.15.0-46-generic
2019/03/14 08:34:53 [notice] 5542#0: getrlimit(RLIMIT_NOFILE): 1024:1048576
2019/03/14 08:34:53 [notice] 5560#0: start worker processes
2019/03/14 08:34:53 [notice] 5560#0: start worker process 5561
2019/03/14 08:34:53 [notice] 5560#0: start worker process 5562
2019/03/14 08:34:53 [notice] 5560#0: start worker process 5563
2019/03/14 08:34:53 [notice] 5560#0: start worker process 5564
2019/03/14 08:34:53 [error] 5561#0: *7 [lua] license_helpers.ljbc:0: (): The Kong Enterprise license will expire on 2019-04-11. Please contact <support@konghq.com> to renew your license., context: ngx.timer
2019/03/14 08:37:51 [notice] 5560#0: signal 15 (SIGTERM) received from 5646, exiting
2019/03/14 08:37:51 [notice] 5563#0: exiting
2019/03/14 08:37:51 [notice] 5561#0: exiting
2019/03/14 08:37:51 [notice] 5564#0: exiting
2019/03/14 08:37:51 [notice] 5562#0: exiting
2019/03/14 08:37:51 [notice] 5561#0: exit
2019/03/14 08:37:51 [notice] 5563#0: exit
2019/03/14 08:37:51 [notice] 5560#0: signal 17 (SIGCHLD) received from 5561
2019/03/14 08:37:51 [notice] 5560#0: worker process 5561 exited with code 0
2019/03/14 08:37:51 [notice] 5560#0: signal 29 (SIGIO) received
2019/03/14 08:37:51 [notice] 5560#0: signal 17 (SIGCHLD) received from 5563
2019/03/14 08:37:51 [notice] 5560#0: worker process 5563 exited with code 0
2019/03/14 08:37:51 [notice] 5560#0: signal 29 (SIGIO) received
2019/03/14 08:37:51 [notice] 5564#0: exit
2019/03/14 08:37:51 [notice] 5560#0: signal 17 (SIGCHLD) received from 5564
2019/03/14 08:37:51 [notice] 5560#0: worker process 5564 exited with code 0
2019/03/14 08:37:51 [notice] 5560#0: signal 29 (SIGIO) received
2019/03/14 08:37:51 [notice] 5562#0: exit
2019/03/14 08:37:51 [notice] 5560#0: signal 17 (SIGCHLD) received from 5562
2019/03/14 08:37:51 [notice] 5560#0: worker process 5562 exited with code 0
2019/03/14 08:37:51 [notice] 5560#0: exit
2019/03/14 08:37:54 [notice] 5667#0: using the "epoll" event method
2019/03/14 08:37:54 [notice] 5667#0: openresty/1.13.6.2

- Kong configuration (the output of a GET request to Kong's Admin port - see
  https://docs.konghq.com/latest/admin-api/#retrieve-node-information)
- Operating system
ubuntu-18.04.2-live-server-amd64

**Comment by Jelts** [comment]
root@ubuntu:~# kong start
2019/03/20 03:47:39 [warn] ulimit is currently set to "1024". For better perform                                                                                                             ance set it to at least "4096" using "ulimit -n"
Error: /usr/local/openresty/site/lualib/kong/cmd/start.ljbc:0: nginx: [error] in                                                                                                             it_by_lua error: /usr/local/openresty/site/lualib/kong/core/handler.ljbc:0: atte                                                                                                             mpt to compare number with userdata
stack traceback:
        /usr/local/openresty/site/lualib/kong/core/handler.ljbc: in function </u                                                                                                             sr/local/openresty/site/lualib/kong/core/handler.ljbc:0>
        [C]: in function ''
        /usr/local/openresty/site/lualib/kong/core/handler.ljbc: in function 'bu                                                                                                             ild_router'
        /usr/local/openresty/site/lualib/kong/init.ljbc: in function 'init'
        init_by_lua:3: in main chunk


  Run with --v (verbose) or --vv (debug) for more details

**Comment by p0pr0ck5** [comment]
Hi @Jelts,

If you are using Kong Enterprise, please visit the Enterprise support portal at https://support.konghq.com/support/s/. Thanks!

**Comment by Jelts** [comment]
OK，I see. Thanks

**Comment by Jelts** [comment]
Hi @p0pr0ck5 , I'm a trial user of Kong enterprise. The address you  provided requires sign in but I don't have an account. That page doesn't provide a sign up button. Would you please help me out? Thanks.

**Comment by Jelts** [comment]
Hi @p0pr0ck5 , I'm a trial user of Kong enterprise. The address you provided requires sign in but I don't have an account. That page doesn't provide a sign up button. Would you please help me out? Thanks.

**Comment by p0pr0ck5** [comment]
Hey, sorry about the confusion.

Can you detail how you created the Route objects? It sounds possibly like a Route was created with a nil regex priority.

**Comment by Jelts** [comment]
Hi, this is the code with which I created the last Route object as soon as the 'All Route objects disappeared' thing happened, and it was the source code of the 'Add Route'-Operation on kong manager(I didn't type this manually, I type key-'F12' and found it in the browser).
{
	"protocols": ["http"],
	"methods": ["GET"],
	"hosts": ["www.baidustrip.com"],
	"paths": ["/request"],
	"regex_priority": null,
	"strip_path": true,
	"preserve_host": false,
	"service": {
		"id": "85f79097-217f-4d73-966c-06a328c44da5"
	}
}
As for the regex priority, isn't it a optional parameter? I refered to the document at https://docs.konghq.com/1.0.x/admin-api/#add-route.

Thanks a lot for your reply, my company is considering to adopt kong to replace the existing gateway and I am one of the members who is sent to survey if kong can be qualified.

**Comment by Jelts** [comment]
Hi @p0pr0ck5 , this is the code with which hardly had I created the last Route object when the 'All Route objects disappeared' thing happened, and it was the source code of the 'Add Route'-Operation on kong manager(I didn't type this manually, I type key-'F12' and found it in the browser).
{
"protocols": ["http"],
"methods": ["GET"],
"hosts": ["www.baidustrip.com"],
"paths": ["/request"],
"regex_priority": null,
"strip_path": true,
"preserve_host": false,
"service": {
"id": "85f79097-217f-4d73-966c-06a328c44da5"
}
}
As for the regex priority, isn't it a optional parameter? I refered to the document at https://docs.konghq.com/1.0.x/admin-api/#add-route.

My company is considering to adopt kong to replace the existing gateway and I am one of the members who is sent to survey if kong could be qualified. I would be really appreciative if you helped me out.
